pkg/generate/app/pipeline.go

```diff
@@ -190,6 +190,10 @@ func (s sortablePorts) Swap(i, j int) {
 	s[j] = p
 }
 
+func portKey(port int, protocol kapi.Protocol) string {
+	return fmt.Sprintf("%d/%s", port, protocol)
```

Comment:
it really doesn't matter, it's just to make sure that a port is unique... in the Dockerfile, there's specified as [num]/[protocol] and that's what I followed here.